### PR TITLE
CSES Sliding Window Median - Java solution using Red-Black Order Statistics Tree

### PR DESCRIPTION
**Pull request by andreionoie**
Since there is no "official" order statistics library in Java and the code is not insanely long, we can fit an implementation along with the solution. Instead of storing Pairs of <int, int>, we can modify the behaviour of each node to store the count of that key and correctly handle it when updating the `size` field (which is used in the order statistics operations). This is a little bit more efficient than storing each duplicate node separately along with its index.

Code is directly adapted from CLRS Introduction to Algorithms 3rd ed.

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Comment by andreionoie** [comment]
I guess it would be nice to have a separate page where we present Java alternatives to GNU PBDS or other cpp-native structures (e.g. multiset).

**Comment by SansPapyrus683** [comment]
that's a good idea actually!

also uh i'm gonna need a screenshot of AC for merge

**Comment by andreionoie** [comment]
Had to use an optimized `nextInt()` since the normal one from Kattio gave TLE:
<details><summary>Screenshot</summary>



</details>

**Comment by SansPapyrus683** [comment]
hm it seems your code here still uses the normal kattio stuff

**Comment by andreionoie** [comment]
Well yes, I thought it'd be more relevant to use the standard Kattio and have 2 test cases TLE instead of cluttering the code even more. (kinda sucks that you need to code the most optimized I/O for many CSES problems in Java)

**Comment by SansPapyrus683** [comment]
hm if that's the case add a warning tag

File: solutions/gold/cses-1076.mdx [hunk 1]
```diff
@@ -191,4 +191,339 @@ int main() {
...
+
+			io.print(" " + slidingWindow.findByOrder(medianRank));
+		}
+
+		io.close();
+	}
+
+	// CodeSnip{Kattio}
+
+	static class OrderedMultiset<K extends Comparable<K>> {
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1076.mdx:244]: codesnip this maybe?

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**